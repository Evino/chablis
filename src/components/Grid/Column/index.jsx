import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import { sizes, breakpoints } from '../../../styles/variables';

const ColumnDiv = styled.div`
  width: ${props => props.mobile * 8.33}%;
  margin-left: ${props => props.offsetMobile * 8.33}%;
  max-width: 100%;
  padding: 0 ${sizes.HALF};
  box-sizing: border-box;
  align-self: ${props => props.align};

  @media(min-width: ${breakpoints.EXTRASMALL}) {
    ${props => props.tablet && css`
      width: ${props.tablet * 8.333}%;
    `}

    ${props => props.offsetTablet && css`
     margin-left: ${props.offsetTablet * 8.333}%;
    `}
  }

  @media(min-width: ${breakpoints.NORMAL}) {
    ${props => props.desktop && css`
      width: ${props.desktop * 8.333}%;
    `}

    ${props => props.offsetDesktop && css`
     margin-left: ${props.offsetDesktop * 8.333}%;
    `}
  }
`;

export default class Column extends PureComponent {
  static propTypes = {
    children: PropTypes.node.isRequired,
    mobile: PropTypes.number.isRequired,
    tablet: PropTypes.number,
    desktop: PropTypes.number,
    offsetMobile: PropTypes.number,
    offsetTablet: PropTypes.number,
    offsetDesktop: PropTypes.number,
    align: PropTypes.oneOf([
      'auto',
      'flex-start',
      'center',
      'flex-end',
      'baseline',
      'stretch',
    ]),
  }

  static defaultProps = {
    align: 'auto',
    offsetMobile: 0,
  }

  render() {
    return (
      <ColumnDiv {...this.props}>{this.props.children}</ColumnDiv>
    );
  }
}
