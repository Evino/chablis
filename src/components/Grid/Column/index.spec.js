import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Column from '.';

const element = <Column mobile={12}><span>Subcomponent</span></Column>;
const component = mount(element);

describe('Column', () => {
  it('should render a column div', () => {
    const tree = renderer
      .create(element)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render children properly', () => {
    const child = component.find('span');
    expect(child).toHaveLength(1);
  });

  it('should have a mobile prop', () => {
    expect(component.prop('mobile')).toBe(12);
  });

  it('should render with default alignment', () => {
    expect(component.prop('align')).toBe('auto');
  });

  it('should render with custom aligment', () => {
    component.setProps({ align: 'center' });
    expect(component.prop('align')).toBe('center');
  });
});
