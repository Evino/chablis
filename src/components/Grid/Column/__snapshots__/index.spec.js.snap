// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Column should render a column div 1`] = `
<div
  className="sc-bdVaJa bCOpbA"
>
  <span>
    Subcomponent
  </span>
</div>
`;
